pbs_light_test.go

```diff
@@ -688,6 +689,23 @@ func TestPanicRecovery(t *testing.T) {
 	recovered(nil, pbsmetrics.AdapterLabels{})
 }
 
+// Prevents #648
+func TestCORSSupport(t *testing.T) {
+	handler := func(w http.ResponseWriter, r *http.Request) {}
+	cors := supportCORS(http.HandlerFunc(handler))
+	rr := httptest.NewRecorder()
+	req, err := http.NewRequest("OPTIONS", "http://some-domain.com/openrtb2/auction", nil)
+	req.Header.Set("Access-Control-Request-Method", "POST")
+	req.Header.Set("Access-Control-Request-Headers", "origin")
+	req.Header.Set("Origin", "https://publisher-domain.com")
+
+	if !assert.NoError(t, err) {
+		return
+	}
+	cors.ServeHTTP(rr, req)
+	assert.NotEqual(t, "*", rr.Header().Get("Access-Control-Allow-Origin"))
+}
```

Comment:
Do we want to test if the `Access-Control-Allow-Origin` matches the `Origin` we set in the request?

Reply:
Yes... very good idea. Done